main.js

```diff
@@ -54,25 +54,33 @@ adapter.on('stateChange', function (id, state) {
         //adapter.log.info('ack is not set!');
         let vmdata = id.split('.')[2];
         let type = vmdata.split('_')[0];
-        let vmname = vmdata.split('_')[1];
-        let command = id.split('.')[3]
-        let vmid, node;
-
-        adapter.log.debug('state changed ' + command + ' : type:  ' + type + ' vmname: ' + vmname);
-        // get vm vid
+        // "if" is not working, could not find out why -> dirty workaround
+        //if (type === 'lxc' || type === 'qemu') {
+            let vmname = vmdata.split('_')[1];
```

Comment:
is this really the same array index for name and node?
Please (even if it was that way before) split twice :-)

Reply:
sounds weird, but it is the same index. To verify it: If you enable the logging in line 50 @ (main.js) and 

1. press "start"-button on a proxmox vm called "myfinevm" listed in iobroker, you'll get as value the variable "id":
_'stateChange proxmox.0.qemu_myfinevm.start'_ 
which leads to the vmname => `vmdata.split('_')[1]; = "myfinevm"`
2. press "shutdown" on the node with hostname 'mybigmachine', you'll get as value the variable "id":
_'stateChange proxmox.0.node_mybigmachine.shutdown'_
which leads to the nodename => `vmdata.split('_')[1]; = "mybigmachine"` 

I tried to catch this a bit more elegant with the out-commented "if"-clause, but my javascript-Google-fu was not good enough to get this working ;-)

Does this answer your question? I actually don't know what you mean by "Please split twice" 

kannst mir auch auf deutsch antworten, dann verstehe ich evtl leichter was du meinst ;-)

Reply:
Update: although Google-Fu did not work, asking a real person did help => the actual commit contains the working _if_-clause